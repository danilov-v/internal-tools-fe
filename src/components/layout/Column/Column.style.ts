import styled from 'styled-components';

type ContainerProps = {
  align: 'center' | 'flex-end' | 'flex-start' | 'normal';
  mb: number;
  ml: number;
  mr: number;
  mt: number;
  pb: number;
  pl: number;
  pr: number;
  pt: number;
};

export const Container = styled.div<ContainerProps>`
  align-items: ${({ align }) => align};
  display: flex;
  flex-direction: column;
  margin-bottom: ${({ mb }) => mb}px;
  margin-left: ${({ ml }) => ml}px;
  margin-right: ${({ mr }) => mr}px;
  margin-top: ${({ mt }) => mt}px;
  padding-bottom: ${({ pb }) => pb}px;
  padding-left: ${({ pl }) => pl}px;
  padding-right: ${({ pr }) => pr}px;
  padding-top: ${({ pt }) => pt}px;
`;
