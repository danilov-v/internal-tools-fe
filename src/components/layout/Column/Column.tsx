import React from 'react';

import * as S from './Column.style';

const Column: React.FC<ColumnProps> = ({
  align = 'normal',
  className = '',
  children = '',
  mb = 8,
  ml = 0,
  mr = 0,
  mt = 8,
  pb = 0,
  pl = 0,
  pr = 0,
  pt = 0,
}) => (
  <S.Container
    align={align}
    className={className}
    mb={mb}
    ml={ml}
    mr={mr}
    mt={mt}
    pb={pb}
    pl={pl}
    pr={pr}
    pt={pt}
  >
    {children}
  </S.Container>
);

type ColumnProps = {
  align?: 'center' | 'flex-end' | 'flex-start' | 'normal';
  className?: string;
  mb?: number;
  ml?: number;
  mr?: number;
  mt?: number;
  pb?: number;
  pl?: number;
  pr?: number;
  pt?: number;
};

export { Column };
