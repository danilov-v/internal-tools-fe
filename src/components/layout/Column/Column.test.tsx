import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import 'jest-styled-components';

import { Column } from './Column';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    ...props,
  };

  return render(<Column {...parsedProps} />);
};

describe('Column component', () => {
  it('matches snapshot with only required props', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('matches snapshot with additional props', () => {
    const component = getComponent({
      align: 'center',
      children: 'children',
      mb: 100,
      pt: 20,
    });

    expect(component.baseElement).toMatchSnapshot();
  });
});
