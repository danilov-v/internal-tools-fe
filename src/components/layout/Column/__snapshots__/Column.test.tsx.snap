// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component matches snapshot with additional props 1`] = `
<body>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 100px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 8px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 20px;
}

<div>
    <div
      class="c0"
    >
      children
    </div>
  </div>
</body>
`;

exports[`Column component matches snapshot with only required props 1`] = `
.c0 {
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 8px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 8px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
}

<body>
  <div>
    <div
      class="c0"
    />
  </div>
</body>
`;
